src/test/java/org/jenkinsci/plugins/workflow/cps/global/GrapeTest.java

```diff
@@ -0,0 +1,174 @@
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                FileUtils.write(new File(repo.workspace, "vars/one.groovy"),
+                    "@Grab('org.apache.commons:commons-math3:3.4.1')\n" +
+                    "import org.apache.commons.math3.complex.ComplexField\n" +
+                    "def call() {ComplexField.instance.one}");
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("echo(/${one()} + ${one()} = ${one().add(one())}/)", true));
+                story.j.assertLogContains("(1.0, 0.0) + (1.0, 0.0) = (2.0, 0.0)", story.j.buildAndAssertSuccess(p));
+            }
+        });
+    }
+
+    // TODO test transitive dependencies; need to find something in Central which has a dependency not in this plugin’s test classpath and which could be used easily from a script
+
+    @Ignore("TODO ClassNodeResolver.tryAsLoaderClassOrScript swallows the CompilationFailedException and throws a generic GroovyBugError; fixed in Jenkins 2 / Groovy 2")
+    @Test public void nonexistentLibrary() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                FileUtils.write(new File(repo.workspace, "src/pkg/X.groovy"),
+                    "package pkg\n" +
+                    "@Grab('net.nowhere:nonexistent:99.9')\n" +
+                    "static def run() {}");
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("pkg.X.run()", true));
+                story.j.assertLogContains("net.nowhere", story.j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0)));
+            }
+        });
+    }
+
+    @Ignore("TODO ditto")
+    @Test public void nonexistentImport() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                FileUtils.write(new File(repo.workspace, "src/pkg/X.groovy"),
+                    "package pkg\n" +
+                    "@Grab('commons-primitives:commons-primitives:1.0')\n" +
+                    "import net.nowhere.Nonexistent\n" +
+                    "static def run() {new Nonexistent()}");
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("pkg.X.run()", true));
+                story.j.assertLogContains("net.nowhere.Nonexistent", story.j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0)));
+            }
+        });
+    }
+
+    // TODO test alternate Maven repositories
+
+    @Ignore("TODO MissingMethodException: No signature of method: static com.google.common.base.CharMatcher.whitespace() is applicable for argument types: () values: []")
+    @Test public void overrideCoreLibraries() throws Exception {
```

Comment:
Do not see any way this could work, unfortunately, so you are in general pinned to the versions of libraries shipped in Jenkins.